### PR TITLE
New: class-name-casing rule (class-name from TSLint)

### PR DESCRIPTION
**Pull request by j-f1**
Implements https://github.com/nzakas/eslint-plugin-typescript/issues/5#issuecomment-321985325.

### CODE REVIEW

File: lib/rules/class-name-casing.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+    meta: {
+        docs: {
+            description: "Enforces PascalCased class and interface names.",
+            category: "Fill me in",
```
Comment (weirdpattern) [inline lib/rules/class-name-casing.js]: Was this an omission or is this the right category?

Comment (j-f1) [inline lib/rules/class-name-casing.js]: That’s an omission 🙈

Comment (j-f1) [inline lib/rules/class-name-casing.js]: Fixed!

**Comment by weirdpattern** [comment]
Awesome, code LGTM
Please prefix the title with 'New:' so the PR can be easily identified as a rule addition.

**Comment by corbinu** [comment]
LGTM

**Comment by JamesHenry** [comment]
Thanks! The older README.md cleanup PR that went in means this just needs to be rebased before going in

**Comment by j-f1** [comment]
@JamesHenry Done!

**Pull request merged**